module16/test_cases.robot

```diff
@@ -0,0 +1,66 @@
+*** Settings ***
+Library    SeleniumLibrary
+Library    custom_keywords.py
+
+Test Setup     Run Keywords
+...    ${driver} Open Browser    https://www.demoblaze.com/    Chrome 
+...    AND    Set Selenium Speed    0.5s
+...    AND    Maximize Browser Window                  
+Test Teardown    Close All Browsers
+
+*** Variables ***
+${driver}
+${user_login}    gryundelb@gmail.com
+${user_password}    Password1
+
+*** Test Cases ***
+Test Login
+    [Tags]    screenshot
+    Click Login Button
+    Wait Until Element Is Visible    css=#loginusername
+    Wait Until Element Is Visible    css=#loginpassword
+    Log In
+    Wait Until Element Is Visible    css=#logout2
+    ${welcome_message}=    Get Text    css=#nameofuser
+    Should Be Equal As Strings    ${welcome_message}    Welcome ${user_login}
+    Capture Page Screenshot
+
+Test Add Product To Cart
+    [Tags]    screenshot
+    Click Login Button
+    Wait Until Element Is Visible    css=#loginusername
+    Wait Until Element Is Visible    css=#loginpassword
+    Log In
+    Open Monitors Category
+     ${product_info}=    Get Highest Price Monitor
+    Click Add To Cart Button
+    Click Cart Button
+    Wait Until Element Is Visible    css=.success
+    ${cart_product_info}=    Get Product Info
+    Should Be Equal As Strings    ${product_info['name']}    ${cart_product_info['name']}
```

Comment:
it would be better to add descriptive error message as an arument of keywords like Should Be...